contract_scripts_test.go

```diff
@@ -37,3 +39,72 @@ func TestContractScriptUnmarshalBinary(t *testing.T) {
 	require.Error(err)
 	require.Contains(err.Error(), "failed to read storage")
 }
+
+func TestSerializeTransactionParameters(t *testing.T) {
+	require := require.New(t)
+
+	// "do" entrypoint
+	// ---------------
+	// tezos-client rpc post /chains/main/blocks/head/helpers/forge/operations with '{
+	// 	"branch": "BMTiv62VhjkVXZJL9Cu5s56qTAJxyciQB2fzA9vd2EiVMsaucWB",
+	// 	"contents":
+	// 		[ { "kind": "transaction",
+	// 			"source": "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx",
+	// 			"fee": "1266", "counter": "1", "gas_limit": "10100",
+	// 			"storage_limit": "277",  "amount": "0",
+	// 			"destination": "KT1GrStTuhgMMpzbNWKTt7NoXGrYiufrHDYq",
+	// 			"parameters": {"entrypoint": "do", "value": {}} } ]
+	// }'
+	paramsValueBytes, err := hex.DecodeString("000000050200000000")
+	require.NoError(err)
+	paramsValue := tezosprotocol.TransactionParametersValueRawBytes(paramsValueBytes)
+	params := tezosprotocol.TransactionParameters{
+		Entrypoint: tezosprotocol.EntrypointDo,
+		Value:      &paramsValue,
+	}
+	expectedBytes := "02000000050200000000"
+	observedBytes, err := params.MarshalBinary()
+	require.NoError(err)
+	require.Equal(expectedBytes, hex.EncodeToString(observedBytes))
+	reserialized := tezosprotocol.TransactionParameters{}
+	require.NoError(reserialized.UnmarshalBinary(observedBytes))
+	require.Equal(params, reserialized)
+}
+
+func TestSerializeNamedEntrypoint(t *testing.T) {
+	require := require.New(t)
+
+	// misc named entrypoint
+	// ---------------------
+	// tezos-client rpc post /chains/main/blocks/head/helpers/forge/operations with '{
+	// 	"branch": "BMTiv62VhjkVXZJL9Cu5s56qTAJxyciQB2fzA9vd2EiVMsaucWB",
+	// 	"contents":
+	// 		[ { "kind": "transaction",
+	// 			"source": "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx",
+	// 			"fee": "1266", "counter": "1", "gas_limit": "10100",
+	// 			"storage_limit": "277",  "amount": "0",
+	// 			"destination": "KT1GrStTuhgMMpzbNWKTt7NoXGrYiufrHDYq",
+	// 			"parameters": {"entrypoint": "dummy", "value": {}} } ]
+	// }'
```

Comment:
It would be nice to include the hex version of the transaction here like in other examples

Reply:
The hex version is `e655948a282fcfc31b98abe9b37a82038c4c0e9b8e11f60ea0c7b33e6ecc625f6c0002298c03ed7d454a101eb7022bc95f7e5f41ac78f20901f44e950200015ab81204ccd229281b9c462edaf0a43e78075f4600ffff0564756d6d79000000050200000000`